### PR TITLE
Split integration tests into separate build

### LINKED ISSUE #1546: Speed up release process, aka I hate waiting
The biggest reason why our release process is slow are our integration tests. They take about ~18 minutes. Here's a suggestion for how to we can speed up our pull requests and release builds while still using our existing test suite which is great at identifying regressions.

1. Do not run integration tests by default for a pull request unless certain directories are modified. The reviewer can always run the tests with `/azp run porter-integration` if they are relevant to the pull request. We can start with triggering off of changing the integration tests and when the runtime is changed. This should drastically cut down on PR wait time.
2. Always run the integration tests as a separate pipeline, porter-integration, when pull requests are merged to main. We can check the pipeline status in GitHub to know if canary is healthy.
3. When canary is healthy, a maintainer pushes a tag to cut a release. The release process does not re-run tests already run on canary. We just create build artifacts and publish them.

@vdice What do you think?

### PR DESCRIPTION
**Pull request by carolynvs**
# What does this change
* Rename e2e to smoke tests. These should quickly (<3min) check that we didn't drastically break porter.
* Always run the smoke tests for a pull request, but only run the integration tests when files related to the runtime changed.
* You can manually run the integration tests with `/azp run porter-integration`
* Only run the tests for canary releases. Maintainers tagging a release should always check canary is green first.

# What issue does it fix
Closes #1546 

# Notes for the reviewer
I'd also like to see example bundle building split out too eventually.

# Checklist
- [ ] Unit Tests
- [x] Documentation
- [ ] Schema (porter.yaml)

### CODE REVIEW

**Comment by carolynvs** [comment]
The default PR build is now 8 minutes, including time farting around waiting for an agent.

Screen Shot 2021-05-06 at 3 34 20 PM

File: build/azure-pipelines.integration.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    include:
+      - refs/heads/main
+
+# Only test a PR when a smoke test or the runtime has changed
+pr:
+  autoCancel: true # Cancel an outdated build when people push new changes to their PR
+  paths:
+    include:
+      - build/azure-pipelines.integration.yaml
+      - tests/smoke/*
```
[the remark this answers is not in the document]
Comment (carolynvs) [reply build/azure-pipelines.integration.yml:16]: Oops, yeah that was a brain fart. I'm glad you caught that!

**Review by vdice — APPROVED** [review]
LGTM!

**Pull request merged**